app/services/lms_sync_service.py

```diff
@@ -1,28 +1,41 @@
 import asyncio
-from typing import BinaryIO
+from pydantic import BaseModel, root_validator
+from typing import BinaryIO, Any
 from sqlalchemy.orm import Session
 from app.core.config import settings
-from app.services.canvas_service import CanvasService, UpdateCanvasAssignmentBody
-from app.services.course_service import CourseService
-from app.services.ldap_service import LDAPService
-from app.services.assignment_service import AssignmentService
-from app.services.user.student_service import StudentService
-from app.services.user.instructor_service import InstructorService
-from app.models import AssignmentModel, SubmissionModel
-from app.schemas.course import UpdateCourseSchema
-from app.schemas.assignment import UpdateAssignmentSchema
+from app.services import CourseService, CanvasService, LDAPService, AssignmentService, StudentService, InstructorService, UserService, UpdateCanvasAssignmentBody, LDAPUserInfoSchema
+from app.models import AssignmentModel, StudentModel, SubmissionModel, InstructorModel, UserModel
+from app.schemas import UpdateCourseSchema, UpdateAssignmentSchema
 from app.core.exceptions import (
     AssignmentNotFoundException, NoCourseExistsException, 
     UserNotFoundException, LMSUserNotFoundException
 )
 
+class ResourcePair(BaseModel):
```

Comment:
I think this needs a more descriptive name. It's too generic rn, like if you want a different ResourcePair later on you'd have to rename this one.

Reply:
maybe DbLmsResourcePair?

Reply:
It's not used anywhere besides this module, and in the context of the syncing module, I think it's named appropriately. If you feel strongly I'll rename

Reply:
It's fine